oat-sa/extension-pcisample
# Feature/tr 552 make text reader pci to ims compliant

Related to https://oat-sa.atlassian.net/browse/TR-552

- Convert `textReaderInteraction` API to IMS and move under ims directory.
- Adapt build and install scripts to new place.
- Make factory from PCI.
- Bundle runner and creator and add styles into it.
- Inline img assets into PCI data. To be able to do it, I created a new event listener in `itemCreator` to register `beforeSaveProcess` that should be done before save. ~~Companion PR can be found https://github.com/oat-sa/extension-tao-itemqti/pull/1663~~ Merged to develop
- Fix PCI test.
- Avoid widgetRenderer recreate when runtime loads in authoring.

Requires
- [x] https://github.com/oat-sa/extension-tao-itemqti/pull/1663 to wait for inlining process during save.

Test:
- `cd /tao/view/build` and `npx grunt connect:test:keepalive pcisamplestest` and then open http://127.0.0.1:55988/pciSamples/views/js/pciCreator/ims/textReaderInteraction/test/test.html with your port.
- PCI behavior: add some text and images. Known limitation: XML parse error on BE side if images are bigger than 10MB.
- Install `pciSamples` extension and check PCI behavior to make sure install script works fine.
- Update `pciSamples` extension from old version and check PCI behavior to make sure migration script works fine.
- Create test with PCI and run it on nextgen test runner to be sure PCI does not contain TAO platform dependent feature.
